templates/cluster.yml.j2

```diff
@@ -50,7 +57,7 @@ spec:
     - {{ ip }}
 {% endfor %}
   kubernetesVersion: {{ cluster.version | default(kops_default_version) }}
-  masterPublicName: api.{{ cluster.name }}
+  masterPublicName: k8s-api-{{ cluster_name }}.{{ route53_dns_zone_name }}
```

Comment:
This role itself is not aware of `route53_dns_zone_name` as a variable. Adding this would make it fail.